### PR TITLE
dev: add information on simple printf-style debugging

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
Pulled this from a commit dated `Date:   Sat Feb 3 10:13:45 2018 +1100`

**Comment by rmackay9** [comment]
I wonder if we want this information on the landing page.. I think it might be better if it's put into one of lower level pages or perhaps we need a new page or pages.

File: dev/source/docs/debugging.rst [hunk 1]
```diff
@@ -6,6 +6,69 @@ Debugging
...
+
+
+DataFlash Log messages
+......................
+
+The DataFlash singleton can be used to put messages to the DataFlash log:
+
+::
+
+   DataFlash_Class::instance()->Log_Write_MessageF("u-blox %d HW: %s SW: %s",
```
Comment (tridge) [inline dev/source/docs/debugging.rst]: need to use new AP::Write() API

File: dev/source/docs/debugging.rst [hunk 2]
```diff
@@ -6,6 +6,69 @@ Debugging
...
+                                           AP_HAL::micros64(),
+                                           (double)sqrtf(_control_monitor.rms_roll_P),
+                                           (double)sqrtf(_control_monitor.rms_roll_D),
+                                           (double)sqrtf(_control_monitor.rms_pitch_P),
+                                           (double)sqrtf(_control_monitor.rms_pitch_D),
+                                           (double)sqrtf(_control_monitor.rms_yaw));
+
+See DataFlash.h for the definition of this method.  Generally, it's a 4-character name followed by a comma-separated list of column names, followed by a list of list of format specifiers for each column, followed by the data.  Similar definitions allow addition of multiplier and unit information.
+
+fprintf(stderr, ...)
```
Comment (tridge) [inline dev/source/docs/debugging.rst]: we don't do fprintf and stderr now

File: dev/source/docs/debugging.rst [hunk 3]
```diff
@@ -6,6 +6,69 @@ Debugging
...
+
+Everybody's favourite go-to debugging tool.
+
+Where the output of this command goes is dependendant on which hardware you are running on.
+
+In SITL you can see this output in the xterm Window running the ArduPilot binary - or, on Windows, in the log file created in place of that window.
+
+On Linux this output will appear on ArduPilot's controlling TTY - so in whatever terminal you started ArduPilot in.
+
+On NuttX-based ports, the output will appear
```
Comment (tridge) [inline dev/source/docs/debugging.rst]: nuttx gone

**Comment by rmackay9** [comment]
I don't think we should be putting the info on the landing page.  Instead, for example, the send_text should go into the send_text debugging page.

@peterbarker, are you happy to fix this up?  If not we might just close it..

File: dev/source/docs/debugging.rst (hunk 1 above)
Comment (kd0aij) [reply dev/source/docs/debugging.rst]: http://ardupilot.org/dev/docs/code-overview-adding-a-new-log-message.html

**Comment by peterbarker** [comment]
Closing this - many of the options are no longer open, and it's been pointed out the dataflash stuff has been done elsewhere.

**Pull request closed without merging**